src/org/rascalmpl/util/maven/SimpleResolver.java

```diff
@@ -149,12 +159,17 @@ private void downloadJar(Path local, String groupId, String artifactId, String v
     private void downloadArtifact(Path local, String groupId, String artifactId, String version, boolean force) throws UnresolvableModelException {
         // TODO: deal with repository filters etc in settings.xml
         var url = getUrl(local, groupId, artifactId, version);
-        for (var r : availableRepostories) {
-            if (r.repo.getLayout().equals("legacy")) {
+        for (SimpleRepositoryDownloader repoDownloader : availableRepostories) {
+            Repo originalRepo = repoDownloader.getRepo();
+            Mirror mirror = mirrors.get(originalRepo.getId());
+            if (mirror != null) {
+                repoDownloader = new SimpleRepositoryDownloader(new MirrorRepo(mirror, originalRepo.getRepository()), client);
```

Comment:
can we not do this when a new repo registration comes it? So during the `addRepository` call?